### PR TITLE
Term server + programmatic shell

### PR DESCRIPTION
**Pull request by vietj**
This pull requests adds two features to Vert.x Shell:

- a programmatic interactive shell for running / testing commands via a local interface avoiding the need of a network connection
- provide simple term server feature for SSH and Telnet for implementing simple terminal applications

### CODE REVIEW

**Comment by vietj** [comment]
please someone review

File: src/main/generated/io/vertx/rxjava/ext/shell/system/Job.java [hunk 1]
```diff
@@ -42,58 +45,90 @@ public Object getDelegate() {
     return delegate;
   }
 
+  /**
+   * @return the job id
+   * @return
```
Comment (cescoffier) [inline src/main/generated/io/vertx/rxjava/ext/shell/system/Job.java]: 2 returns ?

File: src/main/generated/io/vertx/rxjava/ext/shell/system/Job.java [hunk 2]
```diff
@@ -42,58 +45,90 @@ public Object getDelegate() {
...
+   */
   public int id() { 
     int ret = this.delegate.id();
     return ret;
   }
 
+  /**
+   * @return the job status
+   * @return 
+   */
```
Comment (cescoffier) [inline src/main/generated/io/vertx/rxjava/ext/shell/system/Job.java]: 2 returns ?

File: src/main/generated/io/vertx/rxjava/ext/shell/system/Job.java [hunk 3]
```diff
@@ -42,58 +45,90 @@ public Object getDelegate() {
...
   public JobStatus status() { 
     JobStatus ret = this.delegate.status();
     return ret;
   }
 
+  /**
+   * @return when the job was last stopped
+   * @return 
+   */
```
Comment (cescoffier) [inline src/main/generated/io/vertx/rxjava/ext/shell/system/Job.java]: 2 returns ?

File: src/main/generated/io/vertx/rxjava/ext/shell/system/Job.java [hunk 4]
```diff
@@ -42,58 +45,90 @@ public Object getDelegate() {
...
   public long lastStopped() { 
     long ret = this.delegate.lastStopped();
     return ret;
   }
 
+  /**
+   * @return the execution line of the job, i.e the shell command line that launched this job
+   * @return 
+   */
```
Comment (cescoffier) [inline src/main/generated/io/vertx/rxjava/ext/shell/system/Job.java]: 2 returns ?

File: src/main/generated/io/vertx/rxjava/ext/shell/system/Job.java [hunk 5]
```diff
@@ -42,58 +45,90 @@ public Object getDelegate() {
...
   public String line() { 
     String ret = this.delegate.line();
     return ret;
   }
 
+  /**
+   * @return the current tty this job uses
+   * @return 
+   */
```
Comment (cescoffier) [inline src/main/generated/io/vertx/rxjava/ext/shell/system/Job.java]: 2 returns ?

**Comment by cescoffier** [comment]
Very cool. +1.

**Pull request merged**